### PR TITLE
Figure.plot: Update docstring for "connection" parameter to GMT 6.5

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

Over time, the general syntax for the `connection` (**F**) parameter of `Figure.plot` has changed:

_[GMT 6.2](https://docs.generic-mapping-tools.org/6.2/plot.html#f)_ `-F[c|n|r][a|f|s|r|refpoint]`
_[GMT 6.3](https://docs.generic-mapping-tools.org/6.3/plot.html#f)_ `-F[c|n|p][a|f|s|r|refpoint]`
_[GMT 6.5](https://docs.generic-mapping-tools.org/6.5/plot.html#f)_ `-F[c|n|p][a|r|s|t|refpoint]`

This PR updates the related docstring in the PyGMT documentation to match the GMT 6.5 documentation. I feel this makes sense, as we link to the last GMT version and we use it in the gallery examples, or?

I am wondering whether the last line of the [figure caption](https://docs.generic-mapping-tools.org/6.5/plot.html#id1) in the GMT documentation has to be updated from
"`directives ra, rt, rs, r10/35 and na`" to
"`directives pa, pt, ps, p10/35 and na`"?

**Preview**:
There is currently an issue with buiding the documentation:
```
generating gallery for gallery... 
[2Kgenerating gallery for gallery/maps... [ 17%] choropleth_map.py
Segmentation fault (core dumped)
```
*Edit 2024/13/01*
Excluding the gallery example `chorophleth_map.py` (see [commit](https://github.com/GenericMappingTools/pygmt/pull/2994/commits/686b4a78a69a83351daaac94d0152c05476015f1)) leads to
```generating gallery for gallery... 
[2Kgenerating gallery for gallery/maps... [ 20%] land_and_water.py
[2Kgenerating gallery for gallery/maps... [ 40%] country_polygons.py
[2Kgenerating gallery for gallery/maps... [ 60%] borders.py
[2Kgenerating gallery for gallery/maps... [ 80%] shorelines.py
[2Kgenerating gallery for gallery/maps... [100%] tilemaps.py
[2Kgenerating gallery for gallery/lines... [  9%] vector_styles.py
[2Kgenerating gallery for gallery/lines... [ 18%] decorated_lines.py
[2Kgenerating gallery for gallery/lines... [ 27%] envelope.py
[2Kgenerating gallery for gallery/lines... [ 36%] great_circles.py
[2Kgenerating gallery for gallery/lines... [ 45%] line_custom_cpt.py
[2Kgenerating gallery for gallery/lines... [ 55%] linefronts.py
[2Kgenerating gallery for gallery/lines... [ 64%] linestyles.py
[2Kgenerating gallery for gallery/lines... [ 73%] quoted_lines.py
[2Kgenerating gallery for gallery/lines... [ 82%] roads.py
Segmentation fault (core dumped)
```
So, the issue seem to be unrelated to the `chorophleth_map.py` example (downloading the data).

After removing both `chorophleth_map.py ` and `roads.py` building the documentation is successful.

https://pygmt-dev--2994.org.readthedocs.build/en/2994/api/generated/pygmt.Figure.plot.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
> There is currently an issue with buiding the documentation:
> 
> ```
> generating gallery for gallery... 
> �[2Kgenerating gallery for gallery/maps... [ 17%] choropleth_map.py
> Segmentation fault (core dumped)
> ```
> 
> _Edit 2024/13/01_ Excluding the gallery example `chorophleth_map.py` (see [commit](https://github.com/GenericMappingTools/pygmt/pull/2994/commits/686b4a78a69a83351daaac94d0152c05476015f1)) leads to
> 
> ```
> �[2Kgenerating gallery for gallery/maps... [ 20%] land_and_water.py
> �[2Kgenerating gallery for gallery/maps... [ 40%] country_polygons.py
> �[2Kgenerating gallery for gallery/maps... [ 60%] borders.py
> �[2Kgenerating gallery for gallery/maps... [ 80%] shorelines.py
> �[2Kgenerating gallery for gallery/maps... [100%] tilemaps.py
> �[2Kgenerating gallery for gallery/lines... [  9%] vector_styles.py
> �[2Kgenerating gallery for gallery/lines... [ 18%] decorated_lines.py
> �[2Kgenerating gallery for gallery/lines... [ 27%] envelope.py
> �[2Kgenerating gallery for gallery/lines... [ 36%] great_circles.py
> �[2Kgenerating gallery for gallery/lines... [ 45%] line_custom_cpt.py
> �[2Kgenerating gallery for gallery/lines... [ 55%] linefronts.py
> �[2Kgenerating gallery for gallery/lines... [ 64%] linestyles.py
> �[2Kgenerating gallery for gallery/lines... [ 73%] quoted_lines.py
> �[2Kgenerating gallery for gallery/lines... [ 82%] roads.py
> Segmentation fault (core dumped)
> ```
> 
> So, the issue seem to be unrelated to the `chorophleth_map.py` example (downloading the data).

I also found the same issue in PR https://github.com/GenericMappingTools/pygmt/pull/2992. There are three files that use external resources and removing all of them fixes the ReadTheDocs failure (see https://github.com/GenericMappingTools/pygmt/pull/2992/commits/70d53af4d07c2739af9b47f5bed1d3d23cd66c2b).

File: pygmt/src/plot.py [hunk 1]
```diff
@@ -122,35 +122,35 @@ def plot(  # noqa: PLR0912
         Draw error bars. Full documentation is at
         :gmt-docs:`plot.html#e`.
     connection : str
-        [**c**\|\ **n**\|\ **r**]\
-        [**a**\|\ **f**\|\ **s**\|\ **r**\|\ *refpoint*].
+        [**c**\|\ **n**\|\ **p**]\
```
Comment (seisman) [inline pygmt/src/plot.py:125]: Please take this opportunity to rewrap the docstrings of **the `connection` parameter** to 88 characters.

Comment (yvonnefroehlich) [reply pygmt/src/plot.py:125]: This is done in commit https://github.com/GenericMappingTools/pygmt/pull/2994/commits/9d4475269003e81751af3462f17aa6d95cadae64.

File: pygmt/src/plot.py [hunk 2]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
         Draw error bars. Full documentation is at
         :gmt-docs:`plot.html#e`.
     connection : str
-        [**c**\|\ **n**\|\ **r**]\
-        [**a**\|\ **f**\|\ **s**\|\ **r**\|\ *refpoint*].
-        Alter the way points are connected (by specifying a *scheme*) and
-        data are grouped (by specifying a *method*). Append one of three
-        line connection schemes:
+        [**c**\|\ **n**\|\ **p**]\
+        [**a**\|\ **r**\|\ **s**\|\ **t**\|\ *refpoint*].
```
Comment (seisman) [inline pygmt/src/plot.py:126]:
```suggestion
        [**c**\|\ **n**\|\ **p**][**a**\|\ **r**\|\ **s**\|\ **t**\|\ *refpoint*].
```

File: pygmt/src/plot.py [hunk 3]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
...
+        Alter the way points are connected (by specifying a *scheme*) and data are
+        grouped (by specifying a *method*). Append one of three line connection
+        schemes:
```
Comment (seisman) [inline pygmt/src/plot.py:129]:
```suggestion
        grouped (by specifying a *method*). Append one of three line connection schemes:
```

File: pygmt/src/plot.py [hunk 4]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
...
-          each group.
-        - **s** : Segment headers are honored so each segment is a group;
-          the group reference point is reset to the first point of each
-          incoming segment [Default].
-        - **r** : Same as **s**, but the group reference point is reset
-          after each record to the previous point (this method is only
-          available with the ``connection="r"`` scheme).
+        - **a**: Ignore all segment headers, i.e., let all points belong to a single
+          group, and set group the reference point to the very first point of the
+          first file.
```
Comment (seisman) [inline pygmt/src/plot.py:139]:
```suggestion
          group, and set group the reference point to the very first point of the first
          file.
```

File: pygmt/src/plot.py [hunk 5]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
...
+        - **r**: Segment headers are honored so each segment is a group; the group
+          reference point is reset after each record to the previous point (this
+          method is only available with the ``connection="p"`` scheme).
```
Comment (seisman) [inline pygmt/src/plot.py:142]:
```suggestion
          reference point is reset after each record to the previous point (this method
          is only available with the ``connection="p"`` scheme).
```

File: pygmt/src/plot.py [hunk 6]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
...
+        - **s**: Same as **r**, but the group reference point is reset to the first
+          point of each incoming segment [Default].
+        - **t**: Consider all data in each file to be a single separate group and
+          reset the group reference point to the first point of each group.
```
Comment (seisman) [inline pygmt/src/plot.py:146]:
```suggestion
        - **t**: Consider all data in each file to be a single separate group and reset
          the group reference point to the first point of each group.
```

File: pygmt/src/plot.py [hunk 7]
```diff
@@ -122,37 +122,32 @@ def plot(  # noqa: PLR0912
...
 
-        Instead of the codes **a**\|\ **f**\|\ **s**\|\ **r** you may append
-        the coordinates of a *refpoint* which will serve as a fixed external
-        reference point for all groups.
+        Instead of the codes **a**\|\ **r**\|\ **s**\|\ **t** you may append the
+        coordinates of a *refpoint* which will serve as a fixed external reference
+        point for all groups.
```
Comment (seisman) [inline pygmt/src/plot.py:150]:
```suggestion
        coordinates of a *refpoint* which will serve as a fixed external reference point
        for all groups.
```

**Review by seisman — APPROVED** [review]
Good to merge after add the two examples back.

**Pull request merged**